daemon/cluster/cluster.go

```diff
@@ -239,8 +293,16 @@ func (c *Cluster) startNewNode(forceNewCluster bool, listenAddr, joinAddr, secre
 		reconnectDelay: initialReconnectDelay,
 	}
 	c.node = node
+	c.localAddr = localAddr
+	c.remoteAddr = remoteAddr
 	c.listenAddr = listenAddr
+	c.advertiseAddr = advertiseAddr
 	c.saveState()
+
+	// Do this assignment after saveState so that the result of
+	// the resolution won't be saved persistently.
+	c.localAddr = actualLocalAddr
```

Comment:
> On reconnection it seems to be reusing c.localAddr and still saving it.

I think this case needs to be fixed. I guess I will split the fields in `Cluster` between the local addr that was passed into `startNewNode`, and the one we are actually using.

> I'm not actually sure about saving the remoteAddr on disk and reusing it. It is likely only valid on the time of the request. If I take down that node that was used as a remote for a specific node I don't want it to affect that node.

`remoteAddr` is only ever used to find `localAddr` in the case where `localAddr` is not given. Basically it's used with a UDP socket to query the kernel's routing table and find what local IP we would use when connecting to that node. It doesn't matter if that node still exists, and we won't affect that node in any way.

I think saving `remoteAddr` so that we can late-bind `localAddr` makes sense, but let me know if you prefer a different way.